src/main/java/com/github/lipiridi/spotless/applier/config/SpotlessOnSaveActionInfo.java

```diff
@@ -0,0 +1,103 @@
+// Copyright 2000-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the
+// Apache 2.0 license.
+package com.github.lipiridi.spotless.applier.config;
+
+import com.intellij.ide.actionsOnSave.ActionOnSaveContext;
+import com.intellij.lang.Language;
+import com.intellij.lang.LanguageFormatting;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.fileTypes.FileTypeManager;
+import com.intellij.openapi.fileTypes.LanguageFileType;
+import com.intellij.openapi.util.Key;
+import com.intellij.psi.codeStyle.CodeStyleSettingsProvider;
+import com.intellij.psi.codeStyle.LanguageCodeStyleSettingsProvider;
+import com.intellij.util.containers.ContainerUtil;
+import com.intellij.util.containers.MultiMap;
+import java.util.Collection;
+import java.util.function.Consumer;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class SpotlessOnSaveActionInfo extends SpotlessOnSaveActionInfoBase<SpotlessOnSaveOptions> {
```

Comment:
Seems like it can extend immediately FormatOnSaveActionInfoBase. And in this case SpotlessOnSaveActionInfoBase is not more needed

Reply:
I don't think this would work, please re-check.

Reply:
I agree, also didn't find the way. Static class `StateBase` is protected